tests/preliminary/test_petrification.py

```diff
@@ -0,0 +1,188 @@
+            "implementation_type": "Dummy",
+        },
+        "ValidatorsExitBusOracle": {
+            "contract": interface.OssifiableProxy(contracts.validators_exit_bus_oracle.address),
+            "proxy_type": "OssifiableProxy",
+            "implementation_type": "Dummy",
+        },
+        # pure implementations
+        "WithdrawalQueueImplementation": {
+            "contract": interface.WithdrawalQueueERC721(lido_dao_withdrawal_queue_implementation),
+            "proxy_type": "Implementation",
+            "implementation_type": "Versioned",
+        },
+        "WithdrawalVaultImplementation": {
+            "contract": interface.WithdrawalVault(lido_dao_withdrawal_vault_implementation),
+            "proxy_type": "Implementation",
+            "implementation_type": "Versioned",
+        },
+        "StakingRouterImplementation": {
+            "contract": interface.StakingRouter(lido_dao_staking_router_implementation),
+            "proxy_type": "Implementation",
+            "implementation_type": "Versioned",
+        },
+        "AccountingOracleImplementation": {
+            "contract": interface.AccountingOracle(lido_dao_accounting_oracle_implementation),
+            "proxy_type": "Implementation",
+            "implementation_type": "Versioned",
+        },
+        "ValidatorsExitBusOracleImplementation": {
+            "contract": interface.ValidatorsExitBusOracle(lido_dao_validators_exit_bus_oracle_implementation),
+            "proxy_type": "Implementation",
+            "implementation_type": "Versioned",
+        },
+        "LidoLocatorImplementation": {
+            "contract": interface.LidoLocator(lido_dao_validators_exit_bus_oracle_implementation),
+            "proxy_type": "Implementation",
+            "implementation_type": "Versioned",
+        },
+    }
+
+
+INITIAL_TOKEN_HOLDER = "0x000000000000000000000000000000000000dead"
+
+def test_is_petrified(petrified_implementations):
+    for contract_name, contract_config in petrified_implementations.items():
+        print("Petrified Contract: {0}".format(contract_name))
+        implementation_address = None
+        proxy_type = contract_config["proxy_type"]
+        implementation_type = contract_config["implementation_type"]
+        if proxy_type == "OssifiableProxy":
+            implementation_address = contract_config["contract"].proxy__getImplementation()
+        elif proxy_type == "WithdrawalsManagerProxy" or proxy_type == "AppProxyUpgradeable":
+            implementation_address = contract_config["contract"].implementation()
+        elif proxy_type == "Implementation":
+            implementation_address = contract_config["contract"].address
+        else:
+            assert False, "unsupported proxy_type"
+
+        assert implementation_address
+
+        if implementation_type == "Versioned":
+            # stub interface for versioned inherited contracts
+            versioned = interface.WithdrawalVault(implementation_address)
```

Comment:
why `WithdrawalVault` but `Versioned` here? Would require adding the ABI taken from lido-dao